### PR TITLE
[WIP]: add kepsff

### PR DESCRIPTION
**Pull request by mirca**

### CODE REVIEW

**Comment by mirca** [comment]
@barentsen kepsff is added. ``outfile`` in ``kepdraw`` is now optional.

**Comment by barentsen** [comment]
Sweet! I'm going to review this tonight.

File: pyke/kepsff.py [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+        Print informative messages and warnings to the shell and logfile?
+    logfile = string (optional)
+        Name of the logfile containing error and warning messages.
+    """
+    # log the call
+    hashline = '--------------------------------------------------------------'
+    kepmsg.log(logfile,hashline,verbose)
+    call = ('KEPSFF -- '
+            'infile={}'.format(infile)
+            'outfile={}'.format(outfile)
```
Comment (barentsen) [inline pyke/kepsff.py]: I'm surprised this works for you -- it gives a SyntaxError on Python 3.5 for me. (Which I can fix by adding a "+" in front if each line.

File: pyke/kepsff.py [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+            'cenmethod={}'.format(cenmethod)
+            'stepsize={}'.format(stepsize)
+            'npoly_cxcy={}'.format(npoly_cxcy)
+            'sigma_cxcy={}'.format(sigma_cxcy)
+            'npoly_ardx={}'.format(npoly_ardx)
+            'npoly_dsdt={}'.format(npoly_dsdt)
+            'sigma_dsdt={}'.format(sigma_dsdt)
+            'npoly_arfl={}'.format(npoly_arfl)
+            'sigma_arfl={}'.format(sigma_arfl)
+            'plotres={}'.format(savep)
```
Comment (barentsen) [inline pyke/kepsff.py]: `savep` is still undefined at this point, when I run it.

File: pyke/kepsff.py [hunk 3]
```diff
@@ -0,0 +1,760 @@
...
+        structure as the input file with three content changes. The aperture
+        photometry, DETSAP_FLUX, and detrended photometry, DETSAP_FLUX are
+        corrected for motion-correlated noise. The quality flag column,
+        SAP_QUALITY, includes a new bit-flag of value 131072 that identifies
+        exposures collected during on-board thruster firings, typically
+        occurring on a 6-hr cadence.
+    datacol : str
+        The name of the FITS data column containing detrended flux data (i.e.
+        low-frequency structure removed). If the input file is a product of
+        kepextract then this column is named DETSAP_FLUX.
```
Comment (barentsen) [inline pyke/kepsff.py]: I think DETSAP_FLUX is only created by `kepflatten`, not `kepextract`.

**Comment by mirca** [comment]
@barentsen oh dear, the last commit should have fixed those issues 🍓

File: pyke/kepsff.py (hunk 3 above)
Comment (mirca) [inline pyke/kepsff.py]: fixed in [ca4c2287](https://github.com/KeplerGO/PyKE/commit/ca4c22877d353be358880f78ed68f1e5d5dab7de)

File: pyke/kepsff.py (hunk 1 above)
Comment (mirca) [inline pyke/kepsff.py]: fixed in [1e8821a](https://github.com/KeplerGO/PyKE/commit/1e8821a69b110fb8d2f796fc99190f6dae23073d)

File: pyke/kepsff.py (hunk 2 above)
Comment (mirca) [inline pyke/kepsff.py]: fixed in [1e8821a](https://github.com/KeplerGO/PyKE/commit/1e8821a69b110fb8d2f796fc99190f6dae23073d), thanks!!

**Comment by barentsen** [comment]
@mirca I made a few changes in 9e33011439d2524ac07df53ae0f60922a4194d07 which were needed for me to make `kepsff` run.  The good news is I am now able to run it on a lightcurve, and the lightcurve improves!

**Comment by mirca** [comment]
@barentsen pictures please

**Pull request merged**